### PR TITLE
ci: provide separate macos releases for intel and arm

### PR DESCRIPTION
**Pull request by dundargoc**
This will immensely reduce the complexity required to support both
architectures, reduce overall lines of code and unblock follow-up
simplifications.

### CODE REVIEW

**Review by clason — APPROVED** [review]

**Comment by dundargoc** [comment]
Todo:test releases

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -59,8 +59,12 @@ jobs:
           ./build/bin/nvim --version | head -n 3 >> $GITHUB_OUTPUT
           printf 'END\n' >> $GITHUB_OUTPUT
 
-  macOS:
-    runs-on: macos-14
+  macos:
```
Comment (clason) [inline .github/workflows/release.yml:62]: **nit:** Why that rename? `macOS` is the correct spelling.

**Comment by clason** [comment]
I think that's missing some adjustments to the `publish` step: if you have two different builds, they should be uploaded separately (and marked as `Intel` and `M1`, not `macos-12` and `macos-14`).

Also, we need to adjust the release template to give correct download instructions.

**Comment by dundargoc** [comment]
Ye, I know lol.

File: .github/workflows/release.yml (hunk 1 above)
Comment (dundargoc) [reply .github/workflows/release.yml:62]: I dislike mixing upper case and lower case in code.

Comment (clason) [reply .github/workflows/release.yml:62]: As long as output remains correct, that is fine.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -59,8 +59,17 @@ jobs:
...
+    strategy:
+      fail-fast: false
+      matrix:
+        runner: [ macos-12, macos-14 ]
+        include:
+          - runner: macos-12
+            arch: x86_64
+          - runner: macos-14
+            arch: arm64
```
Comment (clason) [inline .github/workflows/release.yml:71]: These are the default (only) arches for these runners, but it's good to be explicit here 👍

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -72,17 +81,37 @@ jobs:
         run: printf 'CMAKE_BUILD_TYPE=Release\n' >> $GITHUB_ENV
       - if: github.event_name == 'schedule' || (github.event_name == 'workflow_dispatch' && github.event.inputs.tag_name == 'nightly')
         run: printf 'CMAKE_BUILD_TYPE=RelWithDebInfo\n' >> $GITHUB_ENV
-      - name: Build universal binary
-        run: ./.github/scripts/build_universal_macos.sh
+
+      - name: Build deps
+        run: |
+          MACOSX_DEPLOYMENT_TARGET=11.0
```
Comment (clason) [inline .github/workflows/release.yml:87]:
```suggestion
          export MACOSX_DEPLOYMENT_TARGET=11.0
```
?

**Pull request merged**